packages/connectivity/src/scp-cf/jwt.ts

```diff
@@ -264,6 +269,9 @@ export function verifyJwtWithKey(
       if (!decodedToken) {
         return reject('Invalid JWT. Token verification yielded `undefined`.');
       }
+      if (typeof decodedToken === 'string') {
+        return resolve(JSON.parse(decodedToken));
```

Comment:
Let me know, what do you think about this handling.

Reply:
so I assume the return type of the `verify` of the jsonwebtoken lib changed to allow also for string resposne? The question for me would be if a verification is done if a string is returned? String feels like encode JWT, but decoding is needed to verify so I wonder why after verifying a string is returned. Besides that the change is ok.

Reply:
we discussed and junjie investigated the code, that verify method always does the check. Then depending on the options it either returns the string or object. String is useful when you want to use the verified token directly in a request. So the changes are fine I think.

Reply:
Junjie also looked in the jsondweb token lib that the lib returns a parsable string.

Reply:
In case a base64 encoded string is returned the parse could be an issue. So we decided to use the decode method.

Reply:
> In case a base64 encoded string is returned the parse could be an issue. So we decided to use the decode method.

The `decode` only accepts e.g, `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9`, while the string should look like `"\{\"typ\"\: \"JWT\"\}"`. In this case, the `decode` won't work.

Reply:
hm, so it is not solid in either case. So for me it is fine to merge it the way it is, since on CF the type is also correctly set and we will not run in the string case. Also we know that the lib reponds with a parsable string. If we see errors I think we would need to add a `parse` statement within a try catch in our decode method to handle to two cases. If you want you can also add this to our decode method.

Reply:
hm, so it is not solid in either case. So for me it is fine to merge it the way it is, since on CF the type is also correctly set and we will not run in the string case. Also we know that the lib reponds with a parsable string. If we see errors I think we would need to add a `parse` statement within a try catch in our decode method to handle to two cases. If you want you can also add this to our decode method.

Reply:
From the API contract perspective, yes, this is not reliable at all.
As the source code does the decoding for base 64, I'll keep it for now.

```js
  var payload = payloadFromJWS(jwsSig);
  if (header.typ === 'JWT' || opts.json)
    payload = JSON.parse(payload, opts.encoding);
```
```js
function payloadFromJWS(jwsSig, encoding) {
  encoding = encoding || 'utf8';
  var payload = jwsSig.split('.')[1];
  return Buffer.from(payload, 'base64').toString(encoding);
}
```